providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/wasm/EvaluationWasm.java

```diff
@@ -0,0 +1,161 @@
+        try {
+            ClassLoader classLoader = EvaluationWasm.class.getClassLoader();
+            URL directoryUrl = classLoader.getResource("wasm");
+            if (directoryUrl == null) {
+                throw new RuntimeException("Directory not found");
+            }
+            Path dirPath = Paths.get(directoryUrl.toURI());
+            try (val files = Files.list(dirPath)) {
+                return files.filter(path -> path.getFileName().toString().startsWith("gofeatureflag-evaluation")
+                                && (path.getFileName().toString().endsWith(".wasi")
+                                        || path.getFileName().toString().endsWith(".wasm")))
+                        .findFirst()
+                        .map(Path::toFile)
+                        .orElseThrow(
+                                () -> new RuntimeException("No file starting with 'gofeatureflag-evaluation' found"));
+            }
+        } catch (Exception e) {
+            throw new WasmFileNotFound(e);
+        }
+    }
+
+    /**
+     * getProcExitFunc is a function that is called when the WASM module calls
+     * proc_exit. It throws a WasiExitException with the exit code.
+     * By default, the exit code is 0, and it raises an Exception.
+     *
+     * @return a HostFunction that is called when the WASM module calls proc_exit
+     */
+    private HostFunction getProcExitFunc() {
+        return new HostFunction(
+                "wasi_snapshot_preview1",
+                "proc_exit",
+                Collections.singletonList(ValueType.I32),
+                Collections.emptyList(),
+                (instance, args) -> {
+                    if ((int) args[0] != 0) {
+                        throw new WasiExitException((int) args[0]);
+                    }
+                    return null;
+                });
+    }
+
+    /**
+     * preWarmWasm is a function that is called to pre-warm the WASM module
+     * It calls the malloc function to allocate memory for the WASM module
+     * and then calls the free function to free the memory.
+     */
+    public void preWarmWasm() {
+        val message = "".getBytes(StandardCharsets.UTF_8);
+        Memory memory = this.instance.memory();
+        int len = message.length;
+        int ptr = (int) malloc.apply(len)[0];
+        memory.write(ptr, message);
+        this.free.apply(ptr, len);
+    }
+
+    /**
+     * Evaluate is a function that evaluates the feature flag using the WASM module.
+     *
+     * @param wasmInput - the object used to evaluate the feature flag
+     * @return the result of the evaluation
+     */
+    public GoFeatureFlagResponse evaluate(WasmInput wasmInput) {
+        try {
+            // convert the WasmInput object to JSON string
+            val message = Const.SERIALIZE_WASM_MAPPER.writeValueAsBytes(wasmInput);
+            // Store the json string in the memory
+            Memory memory = this.instance.memory();
```

Comment:
as it includes C memory, maybe some load test in unit test can be added, or is too much ?